### ISSUE TITLE
DR | Contestable issues focus

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#72290]
## Value Statement
**_As a_** screen reader user using the review & submit page
**_I want to_** be able to have my attention focused on the appropriate element
**_So that_** I don't need to navigate through the page to find the problem

## Background Context

On the review & submit page, follow these steps to see the problem:
- Edit contestable issues
- Have at least one additional issue (with edit & remove buttons)
- Deselect all contestable issues
- Focus should move to the "You'll need to select an issue" alert at the top of the page
- Now use the "Update page" button
- Focus now shifts to the first "Remove" button; it should return focus to the alert

![focus moved to remove button after update page](https://github.com/department-of-veterans-affairs/va.gov-team/assets/136959/fe0e9abd-08fa-41f7-84fe-266d7418bd81)

## Acceptance Criteria

- [x] With no issues selected, focus must move to the no issues selected alert upon page submission
- [x] Unit tests and/or e2e tests added to check this behavior
- [x] Behavior has been validated and approved by an a11y expert

## Tasks
- [ ] Investigate the problem with focus management
- [ ] Apply fix to ensure focus is on alert
- [ ] Add tests specific to this focus behavior

## Definition of Ready 
- [ ] Clear value description
- [ ] Testable acceptance criteria
- [ ] Accessibility added to acceptance criteria
- [ ] Approved designs attached
- [ ] Sample data provided where appropriate
- [ ] Estimated to fit within the sprint
- [ ] Dependencies and blockers linked

## Definition of Done
- [ ] Meets acceptance criteria
- [ ] Code reviewed (internal)
- [ ] Submitted to staging
- [ ] Reviewed and approved by product and/or design

**Comment by HeatherWidmont** [comment]
@davidakennedy is still researching & we will discuss estimate when he has a clearer picture of a path forward

**Comment by HeatherWidmont** [comment]
@rocio-desantiago to add the estimate that was agreed upon on the spreadsheet, then she will start work on this

**Comment by HeatherWidmont** [comment]
Looks like this will entail more work than originally anticipated, @Mottie & @rocio-desantiago will re-estimate this using the [estimator spreadsheet](https://docs.google.com/spreadsheets/d/1ae9fpz_ThZycIWjmBBMSEuGRGqsvRSz3/edit#gid=965165439)

**Comment by HeatherWidmont** [comment]
@Mottie & @rocio-desantiago will work in this today

**Comment by HeatherWidmont** [comment]
Team will test in parking lot today

**Comment by Mottie** [comment]
What we found with this is:
- Immediately upon unchecking all the contestable issues, the focus is shifted to the alert & the page is scrolled to bring it into view
- Then if you move focus (or blur) the alert, change the scroll position (move the page), then click the update button, the alert should focus & scroll the alert into view.
- Subsequent clicks on the update button will either:
  - Do nothing (only after the [code from this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/27408) is available in staging)
  - Still focus on the "Remove" button
- We can't solve the subsequent clicks problem easily because of the disconnect between the platform code to update the page, and the contestable issues widget code. If we change the widget into a CustomPage (bypass the schemaform), then we would have better control - but this conversion work may take significant amount of time.

**Comment by Mottie** [comment]
We can test again tomorrow after we know the updated code is available in staging.

**Comment by eileen-coforma** [comment]
Validated on Safari with user 233 (Cara)

It now:
- Focuses on alert when no issues are selected
- After you click off alert, scroll down, and hit update page it refocuses on alert
- If you do it again, it does nothing (it's a separate issue mentioned by Robin above)

**Comment by HeatherWidmont** [comment]
@davidakennedy will take a look & if everything looks good we will close the ticket

**Comment by davidakennedy** [comment]
Tested this today, and everything works as expected. Tested with user 228 (Colder).
 
Works as Eileen mentioned:

- Focuses on alert when no issues are selected
- After you click off alert, scroll down, and hit update page it refocuses on alert
- If you do it again, it does nothing (it's a separate issue mentioned by Robin above)

cc: @HeatherWidmont
